### PR TITLE
ly: 0.2.1 -> 0.5.2.1-1

### PR DESCRIPTION
**Pull request by MatthewCroughan**
###### Motivation for this change

Updates `ly` to the latest version from the developer. The `-1` is due to my fork, which actually allows this package to be compiled, since the developer used a non standard method of managing their .gitmodules which `fetchFromGitHub` can't reconcile. https://github.com/nullgemm/ly/pull/279

Currently, this package seems to be unable to read the `.desktop` files for `wayland-sessions` from the `/nix/store` as noted in https://github.com/nullgemm/ly/issues/193. The intent is to have @darthpjb patch/rewrite `ly` to work with NixOS, whilst fixing the code that requires hardening to be disabled.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117646` at 43c0dc3a run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>ly</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

git is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/display-managers/ly/default.nix:13:3:
```
   |
13 |   buildInputs = [ stdenv libxcb git linux-pam ncurses ];
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/display-managers/ly/default.nix:23:3:
```
   |
23 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/display-managers/ly/default.nix [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
+{ stdenv, lib, fetchFromGitHub, linux-pam, git, libxcb, ncurses }:
+
+# This package is intended to be reworked, it is temporarily using a fork since
+# the original developer used a non-standard method of specifying .gitmodules.
+# https://github.com/nullgemm/ly/pull/279
 
 stdenv.mkDerivation rec {
   pname = "ly";
-  version = "0.2.1";
+  version = "master";
```
Comment (zseri) [inline pkgs/applications/display-managers/ly/default.nix:9]: you should use a more concrete version number

Comment (MatthewCroughan) [reply pkgs/applications/display-managers/ly/default.nix]: This is not utilized, see the comment at the top of the file.

Comment (MatthewCroughan) [reply pkgs/applications/display-managers/ly/default.nix]: I have removed it, I forgot to do so :smile:

File: pkgs/applications/display-managers/ly/default.nix [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
+
+# This package is intended to be reworked, it is temporarily using a fork since
+# the original developer used a non-standard method of specifying .gitmodules.
+# https://github.com/nullgemm/ly/pull/279
 
 stdenv.mkDerivation rec {
   pname = "ly";
-  version = "0.2.1";
+
+  hardeningDisable = [ "all" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/display-managers/ly/default.nix:10]: Can you be more selective and only disable the flags that fail?

File: pkgs/applications/display-managers/ly/default.nix [hunk 3]
```diff
@@ -1,20 +1,26 @@
...
+
+  nativeBuildInputs = [ git ];
+
+  buildInputs = [ stdenv libxcb linux-pam ncurses ];
```
Comment (SuperSandro2000) [inline pkgs/applications/display-managers/ly/default.nix:14]:
```suggestion
  buildInputs = [ libxcb linux-pam ncurses ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/display-managers/ly/default.nix (hunk 2 above)
Comment (MatthewCroughan) [reply pkgs/applications/display-managers/ly/default.nix:10]: I tried, though I could not figure it out. This package is simply insecure, the intention as stated is to have @darthpjb rewrite this such that `hardeningDisable` is not required at all.

File: pkgs/applications/display-managers/ly/default.nix [hunk 4]
```diff
@@ -1,20 +1,26 @@
...
+# the original developer used a non-standard method of specifying .gitmodules.
+# https://github.com/nullgemm/ly/pull/279
 
 stdenv.mkDerivation rec {
   pname = "ly";
-  version = "0.2.1";
+
+  hardeningDisable = [ "all" ];
+
+  nativeBuildInputs = [ git ];
```
[the remark this answers is not in the document]
Comment (MatthewCroughan) [reply pkgs/applications/display-managers/ly/default.nix:12]: Seems to work as intended.

File: pkgs/applications/display-managers/ly/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/applications/display-managers/ly/default.nix:10]: Thats not great to hear.

File: pkgs/applications/display-managers/ly/default.nix [hunk 5]
```diff
@@ -1,20 +1,26 @@
...
+{ stdenv, lib, fetchFromGitHub, linux-pam, git, libxcb, ncurses }:
+
+# This package is intended to be reworked, it is temporarily using a fork since
+# the original developer used a non-standard method of specifying .gitmodules.
+# https://github.com/nullgemm/ly/pull/279
 
 stdenv.mkDerivation rec {
-  pname = "ly";
-  version = "0.2.1";
+  name = "ly";
```
Comment (SuperSandro2000) [inline pkgs/applications/display-managers/ly/default.nix:8]:
```suggestion
  pname = "ly";
  version = "";
```
version still needs to be set. unstable-XXXX-XX-XX would probably be most fitting here.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/display-managers/ly/default.nix [hunk 6]
```diff
@@ -1,20 +1,27 @@
...
+{ stdenv, lib, fetchFromGitHub, linux-pam, git, libxcb, ncurses }:
+
+# This package is intended to be reworked, it is temporarily using a fork since
+# the original developer used a non-standard method of specifying .gitmodules.
+# https://github.com/nullgemm/ly/pull/279
 
 stdenv.mkDerivation rec {
   pname = "ly";
-  version = "0.2.1";
+  version = "";
```
Comment (SuperSandro2000) [inline pkgs/applications/display-managers/ly/default.nix:9]:
```suggestion
  version = "unstable-2021-03-25";
```

**Pull request closed without merging**